packages/connection/src/Manager.php

```diff
@@ -660,4 +1514,117 @@ public function get_access_token( $user_id = false, $token_key = false, $suppres
 			'external_user_id' => (int) $user_id,
 		);
 	}
+
+	/**
+	 * In some setups, $HTTP_RAW_POST_DATA can be emptied during some IXR_Server paths
+	 * since it is passed by reference to various methods.
+	 * Capture it here so we can verify the signature later.
+	 *
+	 * @param Array $methods an array of available XMLRPC methods.
+	 * @return Array the same array, since this method doesn't add or remove anything.
+	 */
+	public function xmlrpc_methods( $methods ) {
+		$this->raw_post_data = $GLOBALS['HTTP_RAW_POST_DATA'];
+		return $methods;
+	}
+
+	/**
+	 * Resets the raw post data parameter for testing purposes.
+	 */
+	public function reset_raw_post_data() {
+		$this->raw_post_data = null;
+	}
+
+	/**
+	 * Registering an additional method.
+	 *
+	 * @param Array $methods an array of available XMLRPC methods.
+	 * @return Array the amended array in case the method is added.
+	 */
+	public function public_xmlrpc_methods( $methods ) {
+		if ( array_key_exists( 'wp.getOptions', $methods ) ) {
+			$methods['wp.getOptions'] = array( $this, 'jetpack_getOptions' );
+		}
+		return $methods;
+	}
+
+	/**
+	 * Handles a getOptions XMLRPC method call.
+	 *
+	 * @param Array $args method call arguments.
+	 * @return an amended XMLRPC server options array.
+	 */
+	public function jetpack_getOptions( $args ) {
+		global $wp_xmlrpc_server;
+
+		$wp_xmlrpc_server->escape( $args );
+
+		$username = $args[1];
+		$password = $args[2];
+
+		$user = $wp_xmlrpc_server->login( $username, $password );
+		if ( ! $user ) {
+			return $wp_xmlrpc_server->error;
+		}
+
+		$options   = array();
+		$user_data = $this->get_connected_user_data();
+		if ( is_array( $user_data ) ) {
+			$options['jetpack_user_id']         = array(
+				'desc'     => __( 'The WP.com user ID of the connected user' ), // phpcs:ignore WordPress.WP.I18n.MissingArgDomain
```

Comment:
Seems like we could ignore that i18n rule for the entire block of code, instead of doing it several times.

Reply:
I think that's fine; I've also added a `@todo` block to audit whether those even make sense - it seems to me they should be using a textdomain, too. Auditing is not required now, and can be done separately.